conda/gateways/disk/create.py

```diff
@@ -246,3 +254,53 @@ def compile_pyc(python_exe_full_path, py_full_path):
                   "  stderr: %s\n"
                   "  rc: %d", command, stdout, stderr, rc)
         raise RuntimeError()
+
+
+def get_json_content(path_to_json):
+    if isfile(path_to_json):
+        try:
+            with open(path_to_json, "r") as f:
+                json_content = json.load(f)
+        except json.decoder.JSONDecodeError:
+            json_content = {}
+    else:
+        json_content = {}
+    return json_content
+
+
+def create_private_envs_meta(pkg, prefix):
+    # type: (str, str -> ()
+    path_to_conda_meta = join(context.root_prefix, "conda-meta")
+    path_to_private_envs = join(path_to_conda_meta, "private_envs")
+
+    if not isdir(path_to_conda_meta):
+        os.mkdir(path_to_conda_meta)
+
+    private_envs_json = get_json_content(path_to_private_envs)
+    private_envs_json[pkg] = prefix
+    with open(path_to_private_envs, "w") as f:
+        json.dump(private_envs_json, f)
+
+
+def remove_private_envs_meta(pkg):
+    path_to_conda_meta = join(context.root_prefix, "conda-meta")
```

Comment:
Just occurred to me that `context.root_prefix` here is going to be problematic in the multi-user case when the user running conda doesn't have write permissions there.  Luckily, conda will just role back now when it hits the permissions issue.  At some point we'll do a better job anticipating permissions issues and give informative error messages.  Don't worry about it right now though.